recipes/qt/5.x.x/conanfile.py

```diff
@@ -801,6 +789,19 @@ def _getenvpath(var):
         if self._settings_build.os == "Linux" and self.settings.compiler == "clang":
             args += ['QMAKE_CXXFLAGS+="-ftemplate-depth=1024"']
 
+        if self._settings_build.os == "Macos":
+            # On macOS, SIP resets DYLD_LIBRARY_PATH injected by VirtualBuildEnv & VirtualRunEnv.
+            # Qt builds several executables (moc etc) which are called later on during build of
+            # libraries, and these executables link to several external dependencies in requirements().
+            # If these external libs are shared, moc calls fail because its dylib dependencies
```

Comment:
Where did the inspiration for this come from? Any extra insight would be appreciated :) but looks good to me

Reply:
I have this error locally without this workaround: https://github.com/conan-io/conan-center-index/pull/20093#issuecomment-1732658091. Injection of VirtualRunEnv in build scope doesn't work on macOS due to SIP (at some point SIP resets DYLD_LIBRARY_PATH, I don't kno exactly when it happens during build, but it's a fact). Previous `BASH_ENV` workaround doesn't work either depending on terminal (zsh doesn't care for example). So if correct zlib version (matching the one installed by conan) is not installed on system, it fails.
The only workaround I've found is to add path to shared libs in LC_RPATH, SIP can't do anything against that.

Reply:
Actually, it's the same hack than the one in AutotoolsToolchain: https://github.com/conan-io/conan/blob/5f0dc4da2202069d20a59ccbbc17b179aaffed33/conan/tools/gnu/autotoolsdeps.py#L58-L60